internal/statestore/redis.go

```diff
@@ -628,32 +620,27 @@ func (rb *redisBackend) GetAssignments(ctx context.Context, id string, callback
 
 // AddProposedTickets appends new proposed tickets to the proposed sorted set with current timestamp
 func (rb *redisBackend) AddTicketsToIgnoreList(ctx context.Context, ids []string) error {
+	if len(ids) == 0 {
+		return nil
+	}
+
 	redisConn, err := rb.connect(ctx)
 	if err != nil {
 		return err
 	}
 	defer handleConnectionClose(&redisConn)
 
-	err = redisConn.Send("MULTI")
-	if err != nil {
-		redisLogger.WithError(err).Error("failed to pipeline commands for AddTicketsToIgnoreList")
-		return status.Error(codes.Internal, err.Error())
-	}
-
 	currentTime := time.Now().UnixNano()
-	for _, id := range ids {
-		// Index the DoubleArg by value.
-		err = redisConn.Send("ZADD", "proposed_ticket_ids", currentTime, id)
-		if err != nil {
-			redisLogger.WithError(err).Error("failed to append proposed tickets to redis")
-			return status.Error(codes.Internal, err.Error())
-		}
+	cmds := make([]interface{}, 2*len(ids)+1)
+	cmds[0] = "proposed_ticket_ids"
```

Comment:
can we use append() over index manipulation? (other places too - easier to read and possibly less bug prone)